cs_upload.py

```diff
@@ -286,51 +334,99 @@ class S3Client:
     object_storage_url: str
     ca_cert: Optional[str]
 
-    def __init__(self, object_storage_url: str, ca_cert: Optional[str]):
+    def __init__(
+        self,
+        object_storage_url: str,
+        ca_cert: Optional[str],
+        check_ca_cert: bool = True,
+        upload_method: str = "POST",
+    ):
         self.object_storage_url = object_storage_url
         self.ca_cert = ca_cert
+        self.check_ca_cert = check_ca_cert
+        self.upload_method = upload_method
 
-    def upload_to_s3(self, form_data: str, trace_file: str) -> None:
-        fields = loads(form_data)
-
-        fields["success_action_status"] = str(fields["success_action_status"])
-        fields["x-amz-meta-filename"] = basename(trace_file)
-
+    def mime_type(self, trace_file: str) -> str:
         if trace_file.endswith(".cst.gz"):
-            mime_type = "application/gzip"
+            return "application/gzip"
         elif trace_file.endswith(".cst"):
-            mime_type = ""
+            return ""
+        elif trace_file.endswith(".pcap"):
+            return "application/octet-stream"
         else:
             print(
-                "Trace file extension must be either .cst.gz or .cst",
+                "Trace file extension must be either .pcap, .cst.gz or .cst",
```

Comment:
**suggestion**: Add `.pcap.gz`, assuming we support that format (I think I saw one somewhere).